core/domain/activity_jobs_one_off.py

```diff
@@ -235,3 +235,197 @@ def reduce(key, values):
+        """Merge the user ids from the snapshot and put them in the parent
+        collection rights model.
+        """
+        content_dict = (
+            collection_models.CollectionRightsModel.convert_to_valid_dict(
+                snapshot_content_model.content))
+        reconstituted_rights_model = (
+            collection_models.CollectionRightsModel(**content_dict))
+        snapshot_metadata_model = (
+            collection_models.CollectionRightsSnapshotMetadataModel.get_by_id(
+                snapshot_content_model.id))
+        snapshot_metadata_model.mentioned_user_ids = list(
+            set(reconstituted_rights_model.owner_ids) |
+            set(reconstituted_rights_model.editor_ids) |
+            set(reconstituted_rights_model.voice_artist_ids) |
+            set(reconstituted_rights_model.viewer_ids))
+        snapshot_metadata_model.put()
+
+    @staticmethod
+    def _add_exploration_user_ids(snapshot_content_model):
+        """Merge the user ids from the snapshot and put them in the parent
+        exploration rights model.
+        """
+        content_dict = (
+            exp_models.ExplorationRightsModel.convert_to_valid_dict(
+                snapshot_content_model.content))
+        reconstituted_rights_model = (
+            exp_models.ExplorationRightsModel(**content_dict))
+        snapshot_metadata_model = (
+            exp_models.ExplorationRightsSnapshotMetadataModel.get_by_id(
+                snapshot_content_model.id))
+        snapshot_metadata_model.mentioned_user_ids = list(
+            set(reconstituted_rights_model.owner_ids) |
+            set(reconstituted_rights_model.editor_ids) |
+            set(reconstituted_rights_model.voice_artist_ids) |
+            set(reconstituted_rights_model.viewer_ids))
+        snapshot_metadata_model.put()
+
+    @staticmethod
+    def _add_topic_user_ids(snapshot_content_model):
+        """Merge the user ids from the snapshot and put them in the parent
+        topic rights model.
+        """
+        reconstituted_rights_model = topic_models.TopicRightsModel(
+            **snapshot_content_model.content)
+        snapshot_metadata_model = (
+            topic_models.TopicRightsSnapshotMetadataModel.get_by_id(
+                snapshot_content_model.id))
+        snapshot_metadata_model.mentioned_user_ids = set(
+            reconstituted_rights_model.manager_ids)
+        snapshot_metadata_model.put()
+
+    @classmethod
+    def enqueue(cls, job_id, additional_job_params=None):
+        # We can raise the number of shards for this job, since it goes only
+        # over three types of entity class.
+        super(AddMentionedUserIdsContentJob, cls).enqueue(
+            job_id, shard_count=64)
+
+    @classmethod
+    def entity_classes_to_map_over(cls):
+        """Return a list of datastore class references to map over."""
+        return [collection_models.CollectionRightsSnapshotContentModel,
```

Comment:
@Hudda -- didn't you write a lint check to force a line break after '[' here?

Reply:
This check is not implemented yet, I will implement it after CSS PR.